src/providers/delimitedtext/qgsdelimitedtextprovider.cpp

```diff
@@ -652,8 +652,21 @@ void QgsDelimitedTextProvider::scanFile( bool buildIndexes )
 
       if ( couldBeTime[i] && !couldBeDateTime[i] )
       {
-        QTime t = QTime::fromString( value );
-        couldBeTime[i] = t.isValid();
+        // Accept 12:34, 12:34:56 or 12:34:56.789
+        // We do not use QTime::fromString() as it accepts strings like
+        // 01/03/2004 as valid times
+        couldBeTime[i] = value.length() >= 5 &&
```

Comment:
@rouault , I looked into this a white back. Around qt 5.12-5.14, QTime::fromString() became a lot saner and limits the format to those values you have in your comment. It's a problem on windows as we still ship an old Qt 5 version.

There could be an ifdef >= 5.14 here to use Qt.

Reply:
ok, done